front-end/src/components/Map/LocationViewMap.js

```diff
@@ -0,0 +1,110 @@
+import React from "react";
+import { Wrapper } from "@googlemaps/react-wrapper";
+import TableMap from "./TableMap";
+import TableData from "../Tables/TableData";
+import Marker from "./Marker";
+import TablePopup from "./TablePopup";
+import Button from "@mui/material/Button";
+import { CircularProgress } from "@mui/material";
+import axios from "axios";
+
+const render = () => {
+  return (
+    <CircularProgress size="100px" />
+  );
+};
+
+const getTablePosition = tbl => {
+  const coords = tbl.coordinateLocation.coordinates;
+
+  return {
+    lat: coords[1],
+    lng: coords[0]
+  }
+};
+
+const LocationViewMap = () => {
+  const [zoom, setZoom] = React.useState(Number(localStorage.getItem('zoom')) || 3);
+  const [center, setCenter] = React.useState({
+    lat: Number(localStorage.getItem('center-lat')) || 40,
+    lng: Number(localStorage.getItem('center-lng')) || -97,
+  });
+
+  const [currentTableId, setCurrentTableId] = React.useState('');
+  const [tableLocations, setTableLocations] = React.useState([]);
+
+  React.useEffect(() => {
+    localStorage.setItem('zoom', zoom);
+  }, [zoom]);
+
+  React.useEffect(() => {
+    localStorage.setItem('center-lat', center.lat);
+    localStorage.setItem('center-lng', center.lng);
+  }, [center]);
+
+  React.useEffect(() => {
+    async function fetchData() {
+      try {
+        let response = await axios.get("api/tables/all");
+        setTableLocations(response.data);
+      } catch (e) {
+        console.log(e);
+        return [];
+      }
+    }
+    
+    fetchData();
+  }, []);
```

Comment:
`useEffect` without providing any dependencies in the dependency array will result in the effect only being run once.

https://stackoverflow.com/a/58579462

Just wanted to make note of this in case someone else runs into it in the future.

Reply:
I updated this to be a function instead of a hook here: 569bf9a9b8fe904ca0fb53161313c37f16bac6b6

Will probably want to do some refactoring so it searches within the bounds of the map, but for now I think this is okay